### ISSUE TITLE
[Direct Line] How to replace default state client with custom client?

### ISSUE THREAD

**Issue by GauravDhavale** [Microsoft/BotBuilder#4083]
## Bot Info
Bot handle:
App ID: 2489776e-b4a1-4d55-b2ba-c430142c75fe
SDK Platform: Node.js
SDK Version: "botbuilder": "^3.11.0", Node.js: 6.10.x
Active Channels: directLine
Deployment Environment: Azure App Service

## Issue Description
My ChatBot will be a part of customer portal website. I will have an authenticated user communicating over my bot.  So, I have created a client with direct Line channel using simple .NET MVC application. To achieve a Transparent authentication, I want to send user's identity information (access token) to the bot when the user accesses the bot over direct line channel. 
To implement this scenario, I was using a state client in my client application created with direct line channel as shown below: 

```cs
 public ActionResult Index()
        {
            var userId = ClaimsPrincipal.Current.FindFirst(ClaimTypes.NameIdentifier).Value;
            GetTokenViaBootStrap().Wait();

            var botCred = new MicrosoftAppCredentials(
              ConfigurationManager.AppSettings["MicrosoftAppId"],
              ConfigurationManager.AppSettings["MicrosoftAppPassword"]);
            var stateClient = new StateClient(botCred);
            BotState botState = new BotState(stateClient);
            BotData botData = new BotData(eTag: "*");
            botData.SetProperty<string>("UserName", result.UserInfo.GivenName + " " + result.UserInfo.FamilyName);
            botData.SetProperty<string>("Email", result.UserInfo.DisplayableId);
            botData.SetProperty<string>("GraphAccessToken", UserAccessToken);
            botData.SetProperty<string>("TokenExpiryTime", result.ExpiresOn.ToString());

            stateClient.BotState.SetUserDataAsync("directline", userId, botData).Wait();

            var UserData = new UserInformationModel
            {
                UserId = userId,
                UserName = result.UserInfo.GivenName + " " + result.UserInfo.FamilyName
            };
            return View(UserData);
        }
```
As the `StateClient` is **Deprecated**, I want to replace `StateClient` by custom client. As suggested in [this ](https://blog.botframework.com/2017/07/18/saving-state-azure-extensions/) article, I tried Azure Table storage and have updated global.asax. **However, Can someone help me to understand on how I'm going to replace above the state client code?** 

In my MVC application, I don't have any Message activity as I'm using `directLine `channel in JavaScript to access the bot as shown below 

```js
BotChat.App({
        bot: { id: 'my_bot_id', name: 'my_bot_id' },
        resize: 'detect',
        sendTyping: true,  
        user: { id: UserData.UserId},
        directLine: {
            secret: "my_bot_secret"
        }
    }, document.getElementById('my_bot_id'));
```

Basically, I want to send the user identification information to the Bot. I have created my bot using Node JS SDK and I'm testing direct Line channel implementation using basic MVC Application.

**Comment by EricDahlvang** [comment]
Answered here: https://stackoverflow.com/questions/48327229/how-to-store-retrieve-bot-data-in-azure-table-storage-with-directline-channel
